src/runtime/sgx/untrusted/sgx_module.cc

```diff
@@ -177,10 +198,20 @@ void tvm_ocall_packed_func(const char* name,
 
 // Allocates space for return values. The returned pointer is only valid between
 // successive calls to `tvm_ocall_reserve_space`.
-void* tvm_ocall_reserve_space(size_t num_bytes) {
-  static thread_local std::vector<uint64_t> buf;
-  buf.reserve(num_bytes);
-  return buf.data();
+void* tvm_ocall_reserve_space(size_t num_bytes, size_t alignment) {
+  static TVMContext ctx = { kDLCPU, 0 };
+  static thread_local void* buf;
+  static thread_local size_t buf_size;
+  static thread_local size_t buf_align;
+
+  if (buf_size >= num_bytes && buf_align >= alignment) return buf;
+
+  DeviceAPI::Get(ctx)->FreeDataSpace(ctx, buf);
```

Comment:
free only if the buf is not nullptr

Reply:
The [C standard (p313)](http://www.open-std.org/JTC1/SC22/wg14/www/docs/n1124.pdf) specifies that `free(NULL)` takes no action. `FreeDataSpace` should uphold this API.

Reply:
also freeing nullptr isn't without [precedent](https://github.com/dmlc/tvm/blob/master/src/runtime/c_runtime_api.cc#L138) in this codebase :)